compiler/daml-extension/src/extension.ts

```diff
@@ -97,6 +99,43 @@ function getViewColumnForShowResource(): ViewColumn {
     }
 }
 
+function exec(command: string, options: cp.ExecOptions, taskName: String): Promise<{ stdout: string; stderr: string }> {
+    return new Promise<{ stdout: string; stderr: string }>((resolve, reject) => {
+        cp.exec(command, options, (error, stdout, stderr) => {
+            if (error) {
+                reject({ error, stdout, stderr });
+                vscode.window.showErrorMessage("Error Generating visual" + taskName + error)
+            }
+            resolve({ stdout, stderr });
+        });
+    });
+}
+
+async function visualize() {
+    tmp.file(function _tempFileCreated(err, path) {
+        if (err) throw err;
+        let buildCmd = "daml build -o " + path
+        let visualizeCmd = "daml damlc visual " + path
+        let workspaceRoot = vscode.workspace.rootPath;
+        let execOpts = { cwd: workspaceRoot }
+        exec(buildCmd, execOpts, "Daml Build Command").then(_ => {
+            exec(visualizeCmd, execOpts, "Generating dot file").then(res => {
+                if (res.stdout) {
+                    vscode.workspace.openTextDocument({ content: res.stdout, language: "dot" }).then(doc =>
+                        vscode.window.showTextDocument(doc, vscode.ViewColumn.One, true).then(_ => loadPreviewIfAvailable())
+                    )
+                }
+            })
+        });
+    });
+}
+
+function loadPreviewIfAvailable() {
+    if (vscode.extensions.getExtension("EFanZh.graphviz-preview")) {
+        vscode.commands.executeCommand("graphviz.showPreviewToSide")
```

Comment:
As this is something that we don't ship along with vsix bundle an error message would mean we DAML extension would fail, maybe an info message would be better I think. Let me add that